### ISSUE TITLE
More tests for the Gibbs sampler

### ISSUE THREAD

**Issue by yebai** [yebai/Turing.jl#78]
Maybe we should implement some models from the [Stan examples](https://github.com/stan-dev/example-models) repo and compare Stan's result with Turing's result. This would allow us more systematically test our Gibbs sampler and provide some insights to the efficiency of current code.

For a start, we could implement some models in the `basic_estimator` folder and `misc` folder?

**Comment by xukai92** [comment]
Good suggestion! Will do it along with bug fixing.

**Comment by arnim** [comment]
Hi,

great library!
While I'm rather new to it I stumbled upon a small issue

```julia
@model gauss1 begin
    N = length(x)
    lam = ones(Float64,N)
    mu ~ Normal(0, sqrt(1000));
    for i = 1:N
        lam[i] ~ Gamma(.001, .001)
        x[i] ~ Normal(mu, sqrt(1 / (sqrt(lam[i]))))
    end
    mu
end

x = [-27.020, 3.570, 8.191, 9.898, 9.603, 9.945, 10.056]
chain = sample(gauss1,  SMC(300))
```

work. However

```julia
@model gauss2 begin
    N = length(x)
    lam = ones(Float64,N)
    mu ~ Normal(0, sqrt(1000));
    
    for i = 1:N
        lam[i] ~ Gamma(.001, .001)
        x[i] ~ Normal(mu, sqrt(1 / (sqrt(lam[i]))))
    end
    mu, lam
end

chain = sample(gauss2,  SMC(300))
```

gives me

`
MethodError: no method matching mcmcstats(::Array{Array{Float64,1},1})
Closest candidates are:
  mcmcstats{T<:Real}(::Array{T<:Real,1}) at /home/juser/.julia/v0.5/Turing/src/core/io.jl:160
  mcmcstats{T<:Real}(::Array{T<:Real,2}) at /home/juser/.julia/v0.5/Turing/src/core/io.jl:168

 in show(::IOContext{Base.AbstractIOBuffer{Array{UInt8,1}}}, ::Turing.Chain) at /home/juser/.julia/v0.5/Turing/src/core/io.jl:85
 in limitstringmime(::MIME{Symbol("text/plain")}, ::Turing.Chain) at /opt/julia_packages/.julia/v0.5/IJulia/src/inline.jl:25
 in display_dict(::Turing.Chain) at /opt/julia_packages/.julia/v0.5/IJulia/src/execute_request.jl:28
 in execute_request(::ZMQ.Socket, ::IJulia.Msg) at /opt/julia_packages/.julia/v0.5/IJulia/src/execute_request.jl:188
 in eventloop(::ZMQ.Socket) at /opt/julia_packages/.julia/v0.5/IJulia/src/eventloop.jl:8
 in (::IJulia.##13#19)() at ./task.jl:360
`

What would you recommend?

Best
Arnim

**Comment by yebai** [comment]
Hi Arnim, 

there is a small bug in the print function for `Chain`, which stores the samples from a sampling algorithm.  The following code with an extra semi-colon runs fine

```julia
chain = sample(gauss2,  SMC(300)); 
```

We will fix this bug in the next release.

Thanks

**Comment by yebai** [comment]
BTW, the current master branch contains some (major) ongoing  refactoring of the interface, the result should not be trusted too much for the moment. This should gets better in the next new release.

**Comment by arnim** [comment]
Thanks @yebai for your quick response.
For now something like in [stan-models/naive_bayes.jl:25](https://github.com/yebai/Turing.jl/commit/3968f0cd8ea557bf5c908308d470163b8c0d58d6#diff-bb6a6dfe9ac099984378069b8e572a4dR25) should do the trick of inspecting.

Let me know when I can help.
Really like the intuitive way of defining programs.

Best
Arnim

**Comment by xukai92** [comment]
@yebai For more examples in the future, we can have separate files for data, model and inference, which would also make tests for front-end/interface only easy. I adapted my previous 3 Stan models following this. Please check https://github.com/xukai92/Turing-exps/tree/master/stan-models.

**Comment by yebai** [comment]
@xukai92 Thanks Kai. I have added some models myself. I suggest that we keep these example models under the `example-models` folder in the main repo for the moment. It will be easier for us to add and run these models within the main repo.

**Comment by xukai92** [comment]
@yebai Agreed. I've made the update to the main repo too.

**Comment by xukai92** [comment]
Goal for this isuee: make Naive Bayes and LDA work

**Comment by xukai92** [comment]
Closed by PR #162
